File: TheMovieApplicationFrontend-main/MovieApplicationFrontend/src/app/service/user-service/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User } from 'src/app/User';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  private userURL : string = "http://localhost:8084/api/v1/user/";
  constructor(private httpClient: HttpClient) { }

  getUser(userEmail : any) : Observable<User>{
    return this.httpClient.get<User>(this.userURL+`${userEmail}`);
  }

  updateUser(user : User, userEmail : any) : Observable<Object>{
    return this.httpClient.put(this.userURL+`${userEmail}`, user);
  }

  deleteUser(userEmail : any) : Observable<Object>{
    return this.httpClient.delete(this.userURL+`${userEmail}`);
  }

  getAllUser() : Observable<User[]>{
    return this.httpClient.get<User[]>(this.userURL+"all");
  }

  getUserCount() : Observable<number>{
    return this.httpClient.get<number>(this.userURL+"count");
  }
}

File: TheMovieApplicationBackend-main/MovieApplication/UserService/src/main/java/com/stackroute/userservice/service/UserService.java
package com.stackroute.userservice.service;

import com.stackroute.userservice.exception.UserAlreadyExistException;
import com.stackroute.userservice.exception.UserNotFoundException;
import com.stackroute.userservice.model.User;

public interface UserService {

    User addUser(User user) throws UserAlreadyExistException;

    boolean deleteUser(String userEmail) throws UserNotFoundException;

    User updateUser(User user, String userEmail) throws UserNotFoundException;

    User getUser(String userEmail) throws UserNotFoundException;




















}
